### DISCUSSION TITLE
MPI_TYPE_SIZE, MPI_FILE_SET_SIZE

### DISCUSSION THREAD

**Discussion by aph9944** [mpi4py/mpi4py#413] [Q&A] [answered]
Hello,

Do you have an example of how to use MPI_TYPE_SIZE, MPI_FILE_SET_SIZE with python. I tried the following but got an error back
ietypesize = MPI.TYPE.SIZE(MPI.DOUBLE)
ifilesize = nxg*nyb*nzb*3*ietypesize
fh.SET_SIZE(ifilesize)

Thanks,

Anthony

**Comment by dalcinl** [comment, marked as answer]
Most of the MPI API exposed by mpi4py is via methods of classes. Therefore, you have to do:
```
etypesize = MPI.DOUBLE.Get_size()
filesize = nx * ny * nz * 3 * etypesize
fh.Set_size(ifilesize)
```

PS: These methods are used in the testsuite. Although I admit that's not the best source for examples, I don't have anything better to offer. Look for additional online resources in the links a the beginning of https://mpi4py.readthedocs.io/en/stable/tutorial.html.

EDIT: The full `mpi4py.MPI` API reference is here:  https://mpi4py.readthedocs.io/en/stable/reference/mpi4py.MPI.html

**Comment by aph9944** [comment]
Thanks. Just got my fh.Write_all to work! Awesome stuff that mpi4py.
